driver/src/contracts/stablex_contract.rs

```diff
@@ -0,0 +1,138 @@
+#[cfg(test)]
+extern crate mock_it;
+
+use web3::types::{H256, U128, U256};
+
+use crate::error::DriverError;
+
+use super::base_contract::BaseContract;
+
+type Result<T> = std::result::Result<T, DriverError>;
+
+#[allow(dead_code)] // event_loop needs to be retained to keep web3 connection open
+struct StableXContractImpl {
+    contract: BaseContract
+}
+
+#[allow(dead_code)] // event_loop needs to be retained to keep web3 connection open
+impl StableXContractImpl {
+    pub fn new(contract: BaseContract) -> Self {
+        // Should we assert that the contract is indeed a StableX contract?
+        StableXContractImpl {
+            contract
+        }
+    }
+}
+
+pub trait StableXContract {
+    fn get_current_auction_index(&self) -> Result<U256>;
+    // TODO - auction_data should parse and return relevant orders and account balances.
+    fn get_auction_data(&self, _index: u32) -> Result<H256>;
```

Comment:
Let's define the interface the way it's intended to be in the end. This should return `Result<(Vec<Order>, AccountState)>`